Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1736,43 +1736,47 @@ class BrowserViewController: UIViewController,
-            creditCardHelper.foundFieldValues = { [weak self] fieldValues, type, frame in
-                guard let tabWebView = tab.webView,
-                      let type = type,
-                      userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true
-                else { return }
-
-                switch type {
-                case .formInput:
-                    TelemetryWrapper.recordEvent(category: .action,
-                                                 method: .tap,
-                                                 object: .creditCardFormDetected)
-                    self?.profile.autofill.listCreditCards(completion: { cards, error in
-                        guard let cards = cards, !cards.isEmpty, error == nil
-                        else {
-                            return
-                        }
-                        DispatchQueue.main.async {
-                            tabWebView.accessoryView.reloadViewFor(.creditCard)
-                            tabWebView.reloadInputViews()
-                        }
-                    })
-                case .formSubmit:
-                    self?.showCreditCardAutofillSheet(fieldValues: fieldValues)
-                    break
-                }
+        let creditCardHelper = CreditCardHelper(tab: tab)
+        tab.addContentScript(creditCardHelper, name: CreditCardHelper.name())
+        creditCardHelper.foundFieldValues = { [weak self] fieldValues, type, frame in
+            guard let tabWebView = tab.webView,
+                  let type = type,
+                  userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true
+            else { return }
+
+            // FXIOS-7150: Telemetry for form input regardless of feature flag status
+            if type == .formInput {
+                TelemetryWrapper.recordEvent(category: .action,
+                                             method: .tap,
+                                             object: .creditCardFormDetected)
+            }
 
-                tabWebView.accessoryView.savedCardsClosure = {
-                    DispatchQueue.main.async { [weak self] in
-                        // Dismiss keyboard
-                        webView.resignFirstResponder()
-                        // Authenticate and show bottom sheet with select a card flow
-                        self?.authenticateSelectCreditCardBottomSheet(fieldValues: fieldValues,
-                                                                      frame: frame)
+            // Perform autofill operations based on feature flag status
+            guard autofillCreditCardStatus else { return }
+            switch type {
+            case .formInput:
+                self?.profile.autofill.listCreditCards(completion: { cards, error in
+                    guard let cards = cards, !cards.isEmpty, error == nil
+                    else {
+                        return
+                    }
```

Comment:
can be one line in this case
```suggestion
                    guard let cards = cards, !cards.isEmpty, error == nil else { return }
```

Reply:
Oh, nice cleanup!! (pushed)